app/cdash/xml_handlers/upload_handler.php

```diff
@@ -165,6 +167,16 @@ public function endElement($parser, $name)
             // Note: Using stream_filter_append/stream_copy_to_stream is more efficient but
             // return an "invalid byte sequence" on windows
             $rhandle = fopen($this->Base64TmpFilename, 'r');
+
+            // Create tmp file
+            $this->TmpFilename = tempnam(sys_get_temp_dir(), 'cdash_upload');
+            if ($this->TmpFilename === false) {
+                Log::error('Failed to create temporary filename');
+                $this->UploadError = true;
+                return;
+            }
+            chmod($this->TmpFilename, 0o644);
```

Comment:
As far as I am aware, it is generally considered bad practice to use commands like `chmod` in PHP, and many systems disable the command.  Although this isn't the first chmod in the codebase, is feels like poor practice to add another if it's possible to avoid it.

Why can't the file be cached in a Laravel storage directory?  Doing so would allow us to keep more secure file permissions, and keep all of the CDash data in one place.

Would it make sense to look for files in the upload storage directory, and then download them to that directory from the primary web server if they weren't found?

Additionally, are there issues with the overall size of these temporary files growing without bound, potentially leading the worker to run out of space and need a restart?

Reply:
Thanks for the feedback. I just pushed a commit to use Laravel Storage facade instead of `sys_get_temp_dir()`. This should remove the need for the hardcoded `chmod` here as well.

I don't think we need to be too worried about disk space constraints here because these temporary files are created and deleted as they are stored (one at a time).